File: Lab 5/C++/main.cpp
#include "Ellipse.h"
#include "Circle.h"
#include "Figure.h"

using namespace _Figure;
using namespace std;

void polymorphysm(const _Figure::Figure& figure);

int main()
{
    Ellipse ellipse(4.2, 2.2);
    Circle circle(3.1);


    std::cout << std::fixed << std::setprecision(2);

    std::cout << "Ellipse:  ";
    polymorphysm(ellipse);

    std::cout << "Circle:  ";
    polymorphysm(circle);  

    return 0;
}

File: Lab 5/C#/main.cs
using System;
using Artem_Figure;

class Program
{
    //static void Main(string[] args)
    static void Main(string[] args)
    {
        Figure ellipse = new Ellipse(4.2, 2.2);
        Figure circle = new Circle(3.1);

        Console.WriteLine("Ellipse:");
        Polymorphism(ellipse);

        Console.WriteLine("Circle:");
        Polymorphism(circle);
    }

    static void Polymorphism(Figure figure)
    {
        Console.WriteLine($"Area: {figure.Area():F2}, Perimeter: {figure.Perimeter():F2}");// F2 менше знаків
    }
}
